pkg/util/azureclient/typeToCode.go

```diff
@@ -0,0 +1,32 @@
+package azureclient
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import "strings"
+
+// GetAPIVersionFromType gets the APIVersion from a full resource type
+func GetAPIVersionFromType(_type string) string {
+	defaultVer := APIVersions["Microsoft.Compute"]
+	if strings.Contains(_type, "/providers/") {
+		// this is to account for types like this: Microsoft.Network/dnsZones/providers/roleAssignments
+		// that needs to be Microsoft.Authorization not Microsoft.Network
```

Comment:
what are the circumstances where this is needed?

Reply:
i'll remove as see what happens